Integrations/CarbonBlackProtect/CarbonBlackProtect.py

```diff
@@ -0,0 +1,1105 @@
+    :param url_params: url parameters for the request
+    :return: File instance response json
+    """
+    return http_request('GET', '/event', params=url_params)
+
+
+def search_approval_request_command():
+    """
+    Searches for approval requests
+    :return: EntryObject of the approval requests
+    """
+    generic_search_command(
+        search_function=search_approval_request,
+        trans_dict=APPROVAL_REQUEST_TRANS_DICT,
+        hr_title='CarbonBlack Protect Approval Request Search',
+        ec_key='CBP.ApprovalRequest(val.ID === obj.ID)'
+    )
+
+
+def search_approval_request(url_params):
+    """
+    Sends the request for approval request, and returns the result json
+    :param url_params: url parameters for the request
+    :return: Approval request response json
+    """
+    return http_request('GET', '/approvalRequest', params=url_params)
+
+
+def search_file_rule_command():
+    """
+    Searches for file rules
+    :return: EntryObject of the file rules
+    """
+    generic_search_command(
+        search_function=search_file_rule,
+        trans_dict=FILE_RULE_TRANS_DICT,
+        hr_title='CarbonBlack Protect File Rule Search',
+        ec_key='CBP.FileRule(val.ID === obj.ID)'
+    )
+
+
+def search_file_rule(url_params):
+    """
+    Sends the request for file rule, and returns the result json
+    :param url_params: url parameters for the request
+    :return: File rule response json
+    """
+    return http_request('GET', '/fileRule', params=url_params)
+
+
+def get_file_rule_command():
+    """
+    Gets the requested file rule
+    :return: EntryObject of the file catalog
+    """
+    generic_get_command(
+        get_function=get_file_rule,
+        trans_dict=FILE_RULE_TRANS_DICT,
+        hr_title='CarbonBlack Protect File Rule Get for {}'.format(demisto.args().get('id')),
+        ec_key='CBP.FileRule(val.ID === obj.ID)'
+    )
+
+
+def get_file_rule(id):
+    """
+    Sends get file rule request
+    :param id: File rule ID
+    :return: Result json of the request
+    """
+    url = '/fileRule/{}'.format(id)
+    return http_request('GET', url)
+
+
+def delete_file_rule_command():
+    """
+    Deletes the requested file rule
+    :return: EntryObject of the file catalog
+    """
+    args = demisto.args()
+    id = args.get('id')
+    delete_file_rule(id)
+    hr = "File Result {} deleted successfully".format(id)
+    demisto.results(hr)
+
+
+def delete_file_rule(id):
+    """
+    Sends delete file rule request
+    :param id: File rule ID
+    :return: Result of the request
+    """
+    url = SERVER + '/fileRule/{}'.format(id)
+    res = requests.request(
+        'DELETE',
+        url,
+        verify=USE_SSL,
+        headers=HEADERS
+    )
+    return res
+
+
+def update_file_rule_command():
+    """
+    Creates or update a file rule
+    :return: Entry object of the created file analysis
+    """
+    args = demisto.args()
+    raw_res = update_file_rule(args)
```

Comment:
this just passing all demisto params and send them as body.
you should extract here what you need and pass it to update_file_rule.

Reply:
Done.